apps/report-documents-not-received/fields/index.js

```diff
@@ -75,6 +64,92 @@ module.exports = {
     ],
     validate: 'required'
   },
+  'dnr-reference-number': {
+    mixin: 'radio-group',
+    labelClassName: 'govuk-label--s',
+    isPageHeading: true,
+    validate: ['required'],
+    options: [
+      {
+        value: 'dnr-record-number',
+        toggle: 'dnr-record-number',
+        child: 'input-text'
+      },
+      {
+        value: 'dnr-case-id',
+        toggle: 'dnr-case-id',
+        child: 'input-text'
+      },
+      {
+        value: 'dnr-ho-reference-number',
+        toggle: 'dnr-ho-reference-number',
+        child: 'input-text'
+      },
+      {
+        value: 'dnr-payment-reference-number',
+        toggle: 'dnr-payment-reference-number',
+        child: 'input-text'
+      },
+      {
+        value: 'dnr-courier-reference-number',
+        toggle: 'dnr-courier-reference-number',
+        child: 'input-text'
+      }
+    ]
+  },
+  'dnr-record-number': {
+    dependent: {
+      field: 'dnr-reference-number',
+      value: 'dnr-record-number'
+    },
+    className: ['govuk-input', 'govuk-!-width-two-thirds'],
+    validate: [
+      'required',
+      'notUrl'
```

Comment:
We can probably add the inbuilt hof `maxlength` validator as well with max length of 250 characters to prevent users from entering more characters. Otherwise, there won't be any limit.